client-js/app/scripts/app.ts

```diff
@@ -105,38 +105,28 @@ window.addEventListener('WebComponentsReady', () => {
 
   var activity: Activity = null;
 
-  function activate(newActivity: Activity): void {
-    var p: Promise<void> = Promise.resolve(null);
-    if (activity) {
-      p = activity.stop();
-    }
```

Comment:
I'm curious about how this original version works. We make a resolved `Promise`, and then if `activity` tests true we set that promise to the result of `activity.stop()`. Then we start chaining callbacks. I guess the idea is that if `activity` tests false then we want to start chaining callbacks onto a `Promise` which completes immediately (i.e. is already resolved), but if `activity` is true then we have to chain after the `stop()` completes.

Do I have it right?